File: washwise-backend/src/main/java/com/washwise/dto/response/OrderResponse.java
package com.washwise.dto.response;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
public class OrderResponse {
    private String id;
    private UserBasicInfo user;
    private ServiceBasicInfo service;
    private BigDecimal totalPrice;
    private String location;
    private LocalDateTime scheduledDate;
    private LocalDateTime completedDate;
    private String notes;
    private String status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    // Add weightKg for frontend compatibility
    private Double weightKg;

    @Data
    @Builder
    public static class UserBasicInfo {
        private String id;
        private String fullName;
        private String email;
    }

    @Data
    @Builder
    public static class ServiceBasicInfo {
        private String id;
        private String name;
        private String description;
        private String category;
        private BigDecimal price;
        private String duration;
        private String imageUrl;
    }
}

File: washwise-mobile/app/src/main/java/com/washwise/mobile/feature/order/data/OrderResponse.kt
package com.washwise.mobile.feature.order.data

data class OrderResponse(
    val id: String,
    val totalPrice: Double? = null,
    val location: String? = null,
    val scheduledDate: String? = null,
    val completedDate: String? = null,
    val notes: String? = null,
    val status: String? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null,
    val weightKg: Double? = null,
    val service: ServiceInfo? = null,
    val user: UserInfo? = null
) {
    data class ServiceInfo(
        val id: String? = null,
        val name: String? = null,
        val category: String? = null,
        val description: String? = null,
        val price: Double? = null,
        val duration: String? = null,
        val imageUrl: String? = null
    )

    data class UserInfo(
        val id: String? = null,
        val fullName: String? = null,
        val email: String? = null
    )
}
